### PR TITLE
Removing CosmosAsyncItemResponse and updating relevant methods and docs

### PR DESCRIPTION
**Pull request by simplynaveen20**
This PR contains-
1. CosmosAsyncItemResponse -> CosmosItemResponse 
2. Map<Response>AndBlock -> Block<Response>
3. Removed sync keyword from the response api doc
4. Removed  unwanted exclusion from code style

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemResponse.java [hunk 1]
```diff
@@ -1,42 +1,102 @@
...
+
+    CosmosItemResponse(ResourceResponse<Document> response, Class<T> classType) {
+        this.itemClassType = classType;
+        this.responseBodyAsByteArray = response.getBodyAsByteArray();
+        this.resourceResponse = response;
     }
 
     /**
-     * Gets resource.
+     * Gets the resource .
```
Comment (David-Noble-at-work) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemResponse.java:38]: Typo: space between “resource” and period.

**Comment by simplynaveen20** [comment]
> LGTM.
> I think for this PR we should run the full suite.
> @simplynaveen20 @moderakh - thoughts ?

I think this is pretty simple pr and emulator test should capture it , no functionality change

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemResponse.java (hunk 1 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemResponse.java]: done

**Comment by kushagraThapar** [comment]
> > LGTM.
> > I think for this PR we should run the full suite.
> > @simplynaveen20 @moderakh - thoughts ?
> 
> I think this is pretty simple pr and emulator test should capture it , no functionality change

Cool.

**Pull request merged**